bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/AbstractRetryingOperation.java

```diff
@@ -410,15 +418,19 @@ protected CallOptions getOperationCallOptions() {
    *     reads.
    * @return a {@link CallOptions}
    */
-  protected CallOptions getRpcCallOptions() {
-    if (callOptions.getDeadline() != null || isStreamingRead()) {
+  protected CallOptions getRpcAttemptCallOptions() {
+    if (callOptionsFactory.isRpcAttemptTimeoutEnabled(request)) {
+      return callOptionsFactory.createForRpcAttempt(request, operationCallOptions.getDeadline());
+    }
```

Comment:
I think entire method can be just be simplified to `return callOptionsFactory.createForRpcAttempt(request, operationCallOptions)`

Reply:
Refactor is done